File: Creational/AbstractFactory/Java/Structure/ConcreteProductB1.java
/**
 * Concrete implementation of ProductB for family 1
 */
public class ConcreteProductB1 implements ProductB {
    
    @Override
    public void operationB() {
        System.out.println("ConcreteProductB1: Operation B executed");
    }
    
    @Override
    public void collaborateWithA(ProductA productA) {
        System.out.println("ConcreteProductB1: Collaborating with " + productA.getClass().getSimpleName());
    }
}

File: Creational/AbstractFactory/TypeScript/Structure/ConcreteProductB1.ts
import { AbstractProductB } from "./AbstractProductB";
import { AbstractProductA } from "./AbstractProductA";

/**
 * ConcreteProductB1 - Concrete implementation of AbstractProductB
 * Represents Product B from Family 1
 */
export class ConcreteProductB1 extends AbstractProductB {
  operationB(): string {
    return "ConcreteProductB1 operation";
  }

  getFamily(): string {
    return "Family 1";
  }

  collaborateWith(productA: AbstractProductA): string {
    return `ConcreteProductB1 collaborating with ${productA.constructor.name}`;
  }
}
